deck-of-cards-kata/src/main/java/bnymellon/codekatas/deckofcards/custom/collections/HashBag.java

```diff
@@ -72,22 +72,22 @@ public void forEachWithIndex(BiConsumer<? super T, Integer> biConsumer)
     {
         Counter index = new Counter();
         this.backingMap.forEach((key, count) ->
-        {
-            for (int i = 0; i < count; i++)
-            {
-                biConsumer.accept(key, index.getCount());
-                index.increment();
-            }
-        });
+                                {
```

Comment:
This is a not what I would expect. Thoughts?

Reply:
I will fix it later today.